### PR TITLE
Nhaimerl direct indirect integration

### PR DESCRIPTION
**Pull request by NikolasHai**

### CODE REVIEW

File: docs/developer-docs/defi/token_integrations/index.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+Non-replicated calls or “query calls” are executed by a single node in the network. Execution can therefore be synchronous, thus much faster. Because a single node is involved in the execution, the result should not be trusted (since the node can be malicious). Nonetheless, one can build trust in the results of query calls via certificates (signatures) which the ICP network can produce. For example, the hash of the last block in the ICP ledger comes with such a certificate which can be verified using the public key of the ICP network. Together with checking the correct chaining of transactions within the ICP ledger, this guarantees that the chain of transactions fetched via queries is indeed genuine. The details of how the certificates are produced and can be verified are in this companion document [here](/docs/references/http-gateway-protocol-spec).
```
Comment (jessiemongeon1) [inline docs/developer-docs/defi/token_integrations/index.mdx:70]:
```suggestion

```

File: docs/developer-docs/defi/token_integrations/index.mdx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+- Reading the result of update calls
+- Deduplication guarantees of the internet computer protocol
+- Ingress message lifecycle 
+- How to verify the result of calls to canisters
+- API of the token canister(s) & their semantics
+- Deduplication guarantees by the ledger 
+- How to fetch the token ledger & verify it is genuine
+- CBOR (Concise Binary Object Representation)
+- Candid (ICP’s interface description language)
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/defi/token_integrations/index.mdx:65]:
```suggestion

To implement a direct integration, you will need to:

- [Call a method of a canister](#http-call-overview): Calling canister methods is done by constructing a message which specifies among other things the canister, the method and its parameters, and submitting this message to the network. 

- Make a call to a canister: To interact with a canister on the Internet Computer, an ingress message is constructed. This message specifies the target canister, the method to be invoked, and the call's payload. Additional parameters include an ingress expiration time, an optional nonce, and the caller's principal identifier. The API of most canisters, including token ledgers, are defined using a [Candid Interface Definition Language (IDL)](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/candid/candid-concepts). Calls can be executed either replicated, or non-replicated, depending on the endpoint where they are submitted. 

- Read the result of a replicated call: The outcome of a replicated call's execution is temporarily stored in the public ingress history of the respective subnet. To access the result, a read state request is submitted, including the previously generated request ID. Given the asynchronous nature of this execution, results must be actively polled. Results are maintained in the ingress history for a brief period (5 minutes post-execution). Therefore, canisters may provide alternative methods for result retrieval that access the canister's internal state directly, such as ledger canisters offering transaction history and balance book access.
```

File: docs/developer-docs/defi/token_integrations/index.mdx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+As explained above, calls can be executed either replicated, or non-replicated, depending on the endpoint where they are submitted. 
+
+#### Reading the result of a replicated call
+The outcome of a replicated call's execution is temporarily stored in the public ingress history of the respective subnet. To access the result, a read state request is submitted, including the previously generated request ID. Given the asynchronous nature of this execution, results must be actively polled. It's worth noting that results are maintained in the ingress history for a brief period (5 minutes post-execution). Therefore, canisters may provide alternative methods for result retrieval that access the canister's internal state directly, such as ledger canisters offering transaction history and balance book access.
+
```
Comment (jessiemongeon1) [inline docs/developer-docs/defi/token_integrations/index.mdx:78]:
```suggestion


```

File: docs/developer-docs/defi/token_integrations/index.mdx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+#### Support and Libraries
+
+The Internet Computer supports interactions with ICP canisters through official JavaScript/TypeScript ([agent-js](https://github.com/dfinity/agent-js)) and Rust ([agent-rs](https://docs.rs/ic-agent/latest/ic_agent/)) libraries. Additionally, third-party libraries exist for other languages like Java, C, C++, and Go.
+
+### Ledger Canister API
```
Comment (jessiemongeon1) [inline docs/developer-docs/defi/token_integrations/index.mdx:83]:
```suggestion
### Ledger canister API
```

File: docs/developer-docs/defi/token_integrations/_category_.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+label: 'Token Integration Options'
```
Comment (jessiemongeon1) [inline docs/developer-docs/defi/token_integrations/_category_.yml:1]:
```suggestion
label: 'Token integration'
```

**Review by jessiemongeon1 — CHANGES_REQUESTED** [review]

File: docs/developer-docs/defi/token_integrations/index.mdx (hunk 1 above)
Comment (NikolasHai) [reply docs/developer-docs/defi/token_integrations/index.mdx:70]: It is important to point out how the IC works in order to understand why direct integration works the way that it does and why it is much more complex than using rosetta. I wouldn't take expect users that read this guide to be very familiar with the consensus and replication mechanism of the IC.

File: docs/developer-docs/defi/token_integrations/index.mdx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+There are multiple ways of how to integrate with the ICP ledger or ICRC-1 ledgers. Previously, we have presented how users can integrate with the two ledgers using either 
+ICRC Rosetta or ICP Rosetta. While it is commonly recommended to use these two implementations to integrate with the ledgers, it might be interesting for certain developers to
+directly integrate with these ledgers. What a direct integration means, how it can be achieved and how it differs to the integration with Rosetta is topic of this section. 
+
+# ICP architecture overview 
+
+The ICP network comprises a set of (communicating) blockchains that we call subnets. Each of these blockchains hosts a number of smart contracts, called canisters. Communication with the Internet Computer uses the Internet Computer Protocol. The protocol specifies how external parties can call into canisters and how to verify the results. Verification on the internet computer does not use the concept of a validator, and it does not require running a full node. Instead, output visible to the external world is digitally signed and can be verified using the master key of the internet computer.
```
[the remark this answers is not in the document]
Comment (NikolasHai) [reply docs/developer-docs/defi/token_integrations/index.mdx:17]: I would not expect a user to be familiar with the architecture of the IC. It is important to know about the verification mechanism of the IC to understand why the direct integration needs a management of call and read states.

[the remark this answers is not in the document]
Comment (jessiemongeon1) [reply docs/developer-docs/defi/token_integrations/index.mdx:15]: Could we link to the other docs that cover this then? Ideally, we want as little duplication across the docs as possible.

File: docs/developer-docs/defi/token_integrations/index.mdx (hunk 1 above)
Comment (jessiemongeon1) [reply docs/developer-docs/defi/token_integrations/index.mdx:73]: Could we link to the other docs that cover this then? Ideally, we want as little duplication across the docs as possible.

**Pull request merged**